thermostat.cpp

```diff
@@ -517,15 +518,15 @@ void DeRestPluginPrivate::handleThermostatClusterIndication(const deCONZ::ApsDat
                     qint8 mode = attr.numericValue().s8;
                     QString mode_set;
 
-                    mode_set = QString("off");
-                    if ( mode == 0x01 ) { mode_set = QString("auto"); }
-                    if ( mode == 0x03 ) { mode_set = QString("cool"); }
-                    if ( mode == 0x04 ) { mode_set = QString("heat"); }
-                    if ( mode == 0x05 ) { mode_set = QString("emergency heating"); }
-                    if ( mode == 0x06 ) { mode_set = QString("precooling"); }
-                    if ( mode == 0x07 ) { mode_set = QString("fan only"); }
-                    if ( mode == 0x08 ) { mode_set = QString("dry"); }
-                    if ( mode == 0x09 ) { mode_set = QString("sleep"); }
+                    mode_set = QLatin1String("off");
+                    if ( mode == 0x01 ) { mode_set = QLatin1String("auto"); }
+                    else if ( mode == 0x03 ) { mode_set = QLatin1String("cool"); }
+                    else if ( mode == 0x04 ) { mode_set = QLatin1String("heat"); }
+                    else if ( mode == 0x05 ) { mode_set = QLatin1String("emergency heating"); }
+                    else if ( mode == 0x06 ) { mode_set = QLatin1String("precooling"); }
+                    else if ( mode == 0x07 ) { mode_set = QLatin1String("fan only"); }
+                    else if ( mode == 0x08 ) { mode_set = QLatin1String("dry"); }
+                    else if ( mode == 0x09 ) { mode_set = QLatin1String("sleep"); }
```

Comment:
Interesting, might be a good candidate for the fancy get mapped value function (doesn't have to be for this PR).

To streamline code style I'd prefer to remove the spaces in the `if` condition and use cameCase instead snake_case.

```
QString modeSet;

if      (mode == 0x01) { modeSet = QLatin1String("auto"); }
else if (mode == 0x03) { modeSet = QLatin1String("cool"); }
else if (mode == 0x04) { modeSet = QLatin1String("heat"); }
else if (mode == 0x05) { modeSet = QLatin1String("emergency heating"); }
else if (mode == 0x06) { modeSet = QLatin1String("precooling"); }
else if (mode == 0x07) { modeSet = QLatin1String("fan only"); }
else if (mode == 0x08) { modeSet = QLatin1String("dry"); }
else if (mode == 0x09) { modeSet = QLatin1String("sleep"); }
else                   { modeSet = QLatin1String("off"); }
```

Note, only the else branch set's the fallback "off" value, no need to set it every time just to overwrite it :)
This raises the question, is every not checked value really "off"?

Reply:
I had exactly the same thought about the mapping function. Regarding the `off` value, I'd say let's keep that for the moment, shouldn't cause any harm. With the new function, that topic's gone anyway.

Regarding the placement of constants and variables, we could make some decisions on how and where to place them best when we talk next time.

Happy to include the further code beauty.

Reply:
Perfect, thanks :)